app/repositories/NrsLockRepository.scala

```diff
@@ -18,73 +18,55 @@ package repositories
 
 import config.AppConfig
 import models.NrsLock
+import org.mongodb.scala.model.Filters.equal
+import org.mongodb.scala.model.Updates.{combine, set}
+import org.mongodb.scala.model._
 import play.api.Logging
-import play.api.libs.json._
-import play.modules.reactivemongo.ReactiveMongoApi
-import reactivemongo.play.json.collection.JSONCollection
+import uk.gov.hmrc.mongo.MongoComponent
+import uk.gov.hmrc.mongo.play.json.Codecs.toBson
+import uk.gov.hmrc.mongo.play.json.PlayMongoRepository
+
+import java.util.concurrent.TimeUnit
 import javax.inject.Inject
 import scala.concurrent.{ExecutionContext, Future}
 
 
-class NrsLockRepository @Inject()(mongo: ReactiveMongoApi,
-                                  config: AppConfig)
-                                 (implicit ec: ExecutionContext) extends Logging {
-
-  implicit final val jsObjectWrites: OWrites[JsObject] = OWrites[JsObject](identity)
-
-  private val collectionName: String = "nrs-lock"
-
-  private val lockTtl: Int = config.lockTtlInSeconds
+class NrsLockRepository @Inject()(
+                                   mongoComponent: MongoComponent,
+                                   config: AppConfig
```

Comment:
you have added new MongoConfig why you are not using it? or if you dont need you can delete MongoConfig class completly